lib/media/segment_index.js

```diff
@@ -147,7 +147,9 @@ shaka.media.SegmentIndex.prototype.merge = function(references) {
             i == this.references_.length - 1 &&
             j == references.length - 1,
             'This should be an update of the last segment in a period');
-        newReferences.push(r2);
+        var r = new shaka.media.SegmentReference(r1.position,
+            r2.startTime, r2.endTime, r2.getUris, r2.startByte, r2.endByte);
+        newReferences.push(r);
```

Comment:
Because if there are leftover new refs after this loop finishes, the code below will append them while adjusting the position numbers correctly  (see line 165+ below).   So if we increment `i` (ignore old ref), the last ref will be treated as a new ref and that code will handle the renumbering.

I am also suggesting that you don't push anything, **just** increment `i`.  For example:

```js
if (r1.startTime < r2.startTime) {
  ...
} else if (r1.startTime > r2.startTime) {
  ...
} else if (Math.abs(r1.endTime - r2.endTime) > 0.1) {
  // When a period is changed...
  goog.asserts.assert(...);
  i++;  // r2 will be pushed as a new ref by the code below.
} else {
  newReferences.push(r1);
  i++;
  j++;
}
```

But maybe this is too subtle, the old code is fine if that is better.